assets/js/modules/analytics-4/components/common/WebDataStreamNameInput.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Analytics WebDataStreamNameInput component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { useMount } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { isURL } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import {
+	FORM_SETUP,
+	MODULES_ANALYTICS_4,
+	WEBDATASTREAM_CREATE,
+} from '../../datastore/constants';
+import { TextField } from 'googlesitekit-components';
+import VisuallyHidden from '../../../../components/VisuallyHidden';
+import WarningIcon from '../../../../../svg/icons/warning-v2.svg';
+import {
+	isValidPropertyID,
+	isValidWebDataStreamName,
+} from '../../utils/validation';
+
+const { useSelect, useDispatch } = Data;
+
+export default function WebDataStreamNameInput() {
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+	const webDataStreamID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getWebDataStreamID()
+	);
+	const webDataStreamName = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'webDataStreamName' )
+	);
+	const webDataStreamAlreadyExists = useSelect( ( select ) =>
+		isValidPropertyID( propertyID )
+			? select( MODULES_ANALYTICS_4 ).doesWebDataStreamExist(
+					propertyID,
+					webDataStreamName
+			  )
+			: false
+	);
+	const siteURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getReferenceSiteURL()
+	);
+
+	const { setValues } = useDispatch( CORE_FORMS );
+
+	const onChange = useCallback(
+		( { currentTarget } ) => {
+			setValues( FORM_SETUP, { webDataStreamName: currentTarget.value } );
+		},
+		[ setValues ]
+	);
+
+	// Set the default web data stream name.
+	useMount( () => {
+		if ( ! webDataStreamName && isURL( siteURL ) ) {
+			const { hostname } = new URL( siteURL );
+
+			setValues( FORM_SETUP, {
+				webDataStreamName: hostname,
+			} );
+		}
+	} );
+
+	// Bounce if an existing web data stream is selected.
```

Comment:
Nit, we should probably use something more descriptive instead of bounce here. ie .Render nothing/return early.

Reply:
Updated, thanks!